### PR TITLE
Add linter & fix some linter errors

### PR DESCRIPTION
**Pull request by suhasagg**
Bug fixes - bugs detected by golangci-lint in go-loom

### CODE REVIEW

File: crypto/yubihsm.go [hunk 1]
```diff
@@ -162,8 +162,10 @@ func (privKey *YubiHsmPrivateKey) deletePrivKey() {
 	}
 
 	// send command
-	privKey.sessionMgr.SendEncryptedCommand(command)
-
+	_,err =privKey.sessionMgr.SendEncryptedCommand(command)
+    if err != nil{
```
Comment (mattkanwisher) [inline crypto/yubihsm.go]: this doesn't look correct, lets just log if there is an error it is not going to remove the privKeyID

File: plugin/contractpb/contractpb.go [hunk 2]
```diff
@@ -43,8 +43,8 @@ type Context interface {
 	plugin.VolatileAPI
 	StaticContext
 	Set(key []byte, pb proto.Message) error
-	Delete(key []byte)
-	HasPermission(token []byte, roles []string) (bool, []string)
+	Delete(key []byte) error
+	HasPermission(token []byte, roles []string) (bool, []string, error)
```
Comment (mattkanwisher) [inline plugin/contractpb/contractpb.go]: did you update loomchain? why do we need an error returned here

Comment (mattkanwisher) [reply plugin/contractpb/contractpb.go]: loomchain doesn't seem to compile

Comment (suhasagg) [reply plugin/contractpb/contractpb.go]: Have fixed the same.

**Comment by suhasagg** [comment]
Relevant Loomchain PR - https://github.com/loomnetwork/loomchain/pull/963

File: crypto/yubihsm.go (hunk 1 above)
Comment (suhasagg) [reply crypto/yubihsm.go]: Done.

File: crypto/yubihsm.go [hunk 3]
```diff
@@ -162,7 +163,8 @@ func (privKey *YubiHsmPrivateKey) deletePrivKey() {
 	}
 
 	// send command
-	privKey.sessionMgr.SendEncryptedCommand(command)
+	_, err = privKey.sessionMgr.SendEncryptedCommand(command)
+	log.Println(err)
```
Comment (enlight) [inline crypto/yubihsm.go]: Don't log it here, return it.

File: plugin/contract.go [hunk 4]
```diff
@@ -72,7 +72,7 @@ type StaticContext interface {
 	Block() loom.BlockHeader
 	Now() time.Time
 	Message() Message
-	GetEvmTxReceipt([]byte) (types.EvmTxReceipt, error)
+	GetEvmTxReceipt([]byte) (types.EvmTxReceipt,error)
```
Comment (enlight) [inline plugin/contract.go]: Formatting

File: plugin/mock_context.go [hunk 5]
```diff
@@ -96,11 +99,16 @@ func (c *FakeContext) WithAddress(addr loom.Address) *FakeContext {
 	return clone
 }
 
-func (c *FakeContext) CreateContract(contract Contract) loom.Address {
-	addr := createAddress(c.address, c.contractNonce)
+func (c *FakeContext) CreateContract(contract Contract) (loom.Address, error) {
+	addr, err := createAddress(c.address, c.contractNonce)
+	if err != nil {
```
Comment (enlight) [inline plugin/mock_context.go]: Instead of returning the error `panic` instead, then we won't have to change a bunch of test code in `loomchain` for no good reason.

File: plugin/mock_context.go [hunk 6]
```diff
@@ -199,8 +207,8 @@ func (c *FakeContext) ContractAddress() loom.Address {
 	return c.address
 }
 
-func (c *FakeContext) GetEvmTxReceipt([]byte) (ptypes.EvmTxReceipt, error) {
-	return ptypes.EvmTxReceipt{}, nil
+func (c *FakeContext) GetEvmTxReceipt([]byte) (ptypes.EvmTxReceipt,error) {
+	return ptypes.EvmTxReceipt{},nil
```
Comment (enlight) [inline plugin/mock_context.go]: Formatting

File: plugin/plugin.go [hunk 7]
```diff
@@ -171,8 +173,6 @@ type GRPCContext struct {
 	contractAddr loom.Address
 }
 
-var _ Context = &GRPCContext{}
```
Comment (enlight) [inline plugin/plugin.go]: Why was this removed?

File: cli/util.go [hunk 8]
```diff
@@ -30,6 +30,9 @@ func ParseAddress(s string) (loom.Address, error) {
 	}
 
 	b, err := ParseBytes(s)
+	if err != nil {
+		return loom.Address{}, nil
```
Comment (enlight) [inline cli/util.go]: This should return the error, not hide it

File: client/evm_contract.go [hunk 9]
```diff
@@ -41,6 +41,9 @@ func DeployContract(client *DAppChainRPCClient, byteCode []byte, signer auth.Sig
 	}
 	data := vm.DeployResponseData{}
 	err = proto.Unmarshal(response.Output, &data)
+	if err != nil {
+		return nil, []byte{}, err
+	}
```
Comment (enlight) [inline client/evm_contract.go]:
```golang
if err := proto.Unmarshal(response.Output, &data); err != nil {
    return nil, nil, err
}
```

File: plugin/contractpb/servicemap.go [hunk 10]
```diff
@@ -56,6 +56,7 @@ const (
 type serviceMethod struct {
 	method     reflect.Method // receiver method
 	argsType   reflect.Type   // type of the request argument
+	//nolint:unused
```
Comment (enlight) [inline plugin/contractpb/servicemap.go]: If this field is really unused remove it

File: plugin/contractpb/servicemap.go [hunk 11]
```diff
@@ -211,6 +212,7 @@ func isExported(name string) bool {
 	return unicode.IsUpper(rune)
 }
 
+//nolint:deadcode
```
Comment (enlight) [inline plugin/contractpb/servicemap.go]: If this method is really unused remove it

File: auth/secp256k1/secp256k1.go [hunk 12]
```diff
@@ -6,5 +6,6 @@ import (
 
 // Secp256k1Signer implements the Signer interface using secp256k1 keys
 type Secp256k1Signer struct {
+	//nolint:unused
```
Comment (enlight) [inline auth/secp256k1/secp256k1.go]: The correct fix here is to move this `type Secp256k1Signer` to `secp256k1_evm.go` and then add another `type Secp256k1Signer` to `secp256k1_no_evm` that doesn't have this `privateKey` field.
